homeassistant/components/sensor/zamg.py

```diff
@@ -0,0 +1,255 @@
+from homeassistant.const import (
+    TEMP_CELSIUS, CONF_MONITORED_CONDITIONS,
+    CONF_NAME, __version__
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'zamg'
+ATTRIBUTION = 'Data provided by ZAMG'
+
+# Data source only updates once per hour, so throttle to 30min to have
+# reasonably recent data
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+CONF_STATION_ID = "station_id"
+
+VALID_STATION_IDS = (
+    '11010', '11012', '11022', '11035', '11036', '11101', '11121', '11126',
+    '11130', '11150', '11155', '11157', '11171', '11190', '11204'
+)
+
+SENSOR_TYPES = {
+    ATTR_WEATHER_PRESSURE: ('Pressure', 'hPa', 'LDstat hPa', float),
+    'pressure_sealevel': ('Pressure at Sea Level', 'hPa', 'LDred hPa', float),
+    ATTR_WEATHER_HUMIDITY: ('Humidity', '%', 'RF %', int),
+    ATTR_WEATHER_WIND_SPEED: ('Wind Speed', 'km/h', 'WG km/h', float),
+    ATTR_WEATHER_WIND_BEARING: ('Wind Bearing', '°', 'WR °', int),
+    'wind_max_speed': ('Top Wind Speed', 'km/h', 'WSG km/h', float),
+    'wind_max_bearing': ('Top Wind Bearing', '°', 'WSR °', int),
+    'sun_last_hour': ('Sun Last Hour', '%', 'SO %', int),
+    ATTR_WEATHER_TEMPERATURE: ('Temperature', '°C', 'T °C', float),
+    'precipitation': ('Precipitation', 'l/m²', 'N l/m²', int),
+    'dewpoint': ('Dew Point', '°C', 'TP °C', float),
+    # The following probably not useful for general consumption,
+    # but we need them to fill in internal attributes
+    'station_name': ('Station Name', None, 'Name', str),
+    'station_elevation': ('Station Elevation', 'm', 'Höhe m', int),
+    'update_date': ('Update Date', None, 'Datum', str),
+    'update_time': ('Update Time', None, 'Zeit', str),
+}
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_STATION_ID):
+        vol.All(cv.string, vol.In(VALID_STATION_IDS)),
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+# ------------------------------------------------------------------------
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup platform."""
+    station_id = config.get(CONF_STATION_ID)
+    name = config.get(CONF_NAME)
+
+    probe = ZamgData(station_id=station_id, logger=_LOGGER)
+
+    sensors = [ZAMGWeather(probe, variable, name)
+               for variable in config[CONF_MONITORED_CONDITIONS]]
+
+    add_devices(sensors, True)
+
+
+# ------------------------------------------------------------------------
+class ZAMGWeather(Entity):
+    """
+    I am a weather wrapper for a specific station and a specific attribute.
+
+    Multiple instances (one for each condition) will refer to the same
+    probe, so things will only get fetched once.
+    """
+
+    attribution = ATTRIBUTION
+    temperature_unit = TEMP_CELSIUS
```

Comment:
I would like to suggest to avoid intermediate vars. Just use the constants directly.